sources/SilkTouch/SilkTouch/Mods/TransformCOM.cs

```diff
@@ -0,0 +1,749 @@
+                }
+                return base.VisitAssignmentExpression(node);
+            }
+
+            public override SyntaxNode? VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                if (!node.ArgumentList.Arguments.Any(arg => arg.Expression.IsKind(SyntaxKind.NullLiteralExpression)))
+                    return base.VisitInvocationExpression(node);
+
+                var symbolInfo = SemanticModel.GetSymbolInfo(node);
+                var newInvocation = node;
+                bool changed = false;
+                if (symbolInfo.Symbol is IMethodSymbol methodSymbol)
+                {
+                    var arguments = node.ArgumentList.Arguments;
+                    for (int i = 0; i < arguments.Count; i++)
+                    {
+                        var argument = arguments[i];
+                        // Check if the argument is a null literal
+                        if (argument.Expression.IsKind(SyntaxKind.NullLiteralExpression))
+                        {
+                            // Get the corresponding parameter type
+                            var parameterType = methodSymbol.Parameters[i].Type;
+                            var name = GetTypeName(parameterType?.ToString() ?? string.Empty);
+
+                            // Check if the parameter type is ComType
+                            if (ComTypes.Any(type => $"{type.Item1}*" == name))
+                            {
+                                changed = true;
+                                var newDefaultArg = Argument(LiteralExpression(SyntaxKind.DefaultLiteralExpression));
+                                newInvocation = newInvocation.ReplaceNode(argument, newDefaultArg);
+                            }
+                        }
+                    }
+                }
+                return changed ? newInvocation : base.VisitInvocationExpression(node);
+            }
+
+            public override SyntaxNode? VisitBinaryExpression(BinaryExpressionSyntax node)
+            {
+                if (!node.Left.IsKind(SyntaxKind.NullLiteralExpression) && !node.Right.IsKind(SyntaxKind.NullLiteralExpression))
+                    return base.VisitBinaryExpression(node);
+
+                var typeInfoR = SemanticModel.GetTypeInfo(node.Right);
+                var typeInfoL = SemanticModel.GetTypeInfo(node.Left);
+                var nameR = GetTypeName(typeInfoR.Type?.ToString() ?? string.Empty);
+                var nameL = GetTypeName(typeInfoL.Type?.ToString() ?? string.Empty);
+
+
+                if (node.Left.IsKind(SyntaxKind.NullLiteralExpression))
+                {
+                    // Check if the type is a ComType
+                    if (typeInfoR.Type != null && ComTypes.Any(type => $"{type.Item1}*" == nameR))
+                    {
+                        return BinaryExpression(node.Kind(), node.Left, IdentifierName($"{node.Right.ToFullString()}.lpVtbl"));
```

Comment:
Do we have a guarantee the field is lpvtbl? Always is for Win32 but I worry about other nanoCOM-mimicking libraries e.g. the Steam API. May be out of scope, but just want to make sure we know what we're doing by hard coding this and not determining it from the type.

Reply:
this is a name from ClangSharp actually, not designated by com